core/linux/test/TimeZoneNativeTest.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019-2025 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.test
+
+import kotlinx.datetime.IllegalTimeZoneException
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.internal.root
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+class TimeZoneNativeTest {
+
+    @Test
+    fun correctSymlinkTest() = withFakeRoot("${RESOURCES}correct-symlink/") {
+        val tz = TimeZone.currentSystemDefault()
+        assertEquals(TimeZone.of("Europe/Oslo"), tz)
+    }
+
+    @Test
+    fun correctLocaltimeCopyTest() = withFakeRoot("${RESOURCES}correct-localtime-copy/") {
+        val tz = TimeZone.currentSystemDefault()
+        assertEquals(TimeZone.of("Europe/Oslo"), tz)
+    }
+
+    @Test
+    fun fallsBackToUTC() = withFakeRoot("${RESOURCES}falls-back-to-utc/") {
```

Comment:
1. I'd still create this directory and add an empty `.keep` file to git.
2. The name is not very descriptive. `falls-back-to-utc` in what scenarios? The name of the test, the name of the directory, a comment for this test, or all of the above should explain the specific logic: falling back to UTC when there is no `/etc/localtime`.